Client/Frontend/Home/FxHomeRecentlySavedCollectionCell.swift

```diff
@@ -0,0 +1,250 @@
+        let currentItem = recentBookmarks[indexPath.row] as! BookmarkItem
+        let site = Site(url: currentItem.url, title: currentItem.title, bookmarked: true, guid: currentItem.guid)
+        
+        profile?.favicons.getFaviconImage(forSite: site).uponQueue(.main, block: { result in
+            guard let image = result.successValue else { return }
+            
+            cell.heroImage.image = image
+            cell.setNeedsLayout()
+        })
+        
+        cell.bookmarkTitle.text = site.title
+        cell.bookmarkDetails.text = site.secondLevelDomain
+        
+        return cell
+    }
+    
+}
+
+extension FxHomeRecentlySavedCollectionCell: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let bookmark = recentBookmarks[indexPath.row] as! BookmarkItem
+        guard let url = URIFixup.getURL(bookmark.url) else { return }
+        
+        homePanelDelegate?.homePanel(didSelectURL: url, visitType: .bookmark, isGoogleTopSite: false)
+        TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .bookmark, value: .recentlySavedBookmarkCell)
+    }
+}
+
+extension FxHomeRecentlySavedCollectionCell: UICollectionViewDelegateFlowLayout {
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+        return CGSize(width: RecentlySavedCollectionCellUX.cellWidth, height: RecentlySavedCollectionCellUX.cellHeight)
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, insetForSectionAt section: Int) -> UIEdgeInsets {
+        return UIEdgeInsets(top: RecentlySavedCollectionCellUX.generalSpacing, left: 0, bottom: RecentlySavedCollectionCellUX.generalSpacing, right: 0)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        return RecentlySavedCollectionCellUX.generalSpacing
+    }
+    
+}
+
+private struct RecentlySavedBookmarkCellUX {
+    static let generalCornerRadius: CGFloat = 8
+    static let bookmarkTitleFontSize: CGFloat = 17
+    static let bookmarkDetailsFontSize: CGFloat = 12
+    static let labelsWrapperSpacing: CGFloat = 4
+    static let bookmarkStackViewSpacing: CGFloat = 8
+    static let bookmarkStackViewShadowRadius: CGFloat = 4
+    static let bookmarkStackViewShadowOffset: CGFloat = 2
+}
+
+/// A cell used in FxHomeScreen's Recently Saved section.
+class RecentlySavedBookmarkCell: UICollectionViewCell {
+    
+    // MARK: - Properties
+    
+    static let cellIdentifier = "recentlySavedBookmarkCell"
+    
+    // UI
+    var heroImage: UIImageView = {
```

Comment:
nit: `let` over `var`

We also now have a convenience view builder so you can just do
```
let heroImage: UIImageView = .build { imageView in
// view configuration
}
```

and it will initialize the view, set `translatesAutoresizingMaskIntoConstraints` to false automatically and return it :)

Reply:
This is 🔥 Will use it for sure!